pallets/task-scheduler/src/lib.rs

```diff
@@ -99,6 +106,76 @@ pub mod pallet {
 		/// callback transaction. This is likely an internal error.
 		NoLocalAcctForSignedTx,
 	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_initialize(block_number: T::BlockNumber) -> Weight {
+			use sp_io::KillStorageResult;
+
+			log::debug!("Cleaning up expired entries");
+
+			let mut unverified_task_count = 0u32;
+			loop {
+				match PendingTasks::<T>::remove_prefix(block_number, None) {
+					KillStorageResult::SomeRemaining(u) => {
+						unverified_task_count.saturating_accrue(u)
+					},
+					KillStorageResult::AllRemoved(u) => {
+						unverified_task_count.saturating_accrue(u);
+						break;
+					},
+				};
+			}
+
+			<T as Config>::WeightInfo::on_initialize(unverified_task_count)
+		}
+
+		fn offchain_worker(block_number: T::BlockNumber) {
+			let auth_id = match Self::authority_id() {
+				Some(id) => id,
+				None => {
+					log::debug!(target: "task", "Not an authority, skipping offchain work");
+					return;
+				},
+			};
+
+			for (deadline, id, task) in PendingTasks::<T>::iter() {
+				let storage_key = tasks::storage_key(&id);
+				let mut lock = tasks::task_lock::<T>(&storage_key);
+
+				let guard = match lock.try_lock() {
+					Ok(g) => g,
+					Err(_) => continue,
+				};
+
+				log::trace!(target: "task", "@{block_number:?} Task {:8?}", id);
+
+				use tasks::error::TaskError::*;
+				match task.forward_task(deadline) {
+					Ok(call) => {
+						match Self::submit_txn_with_synced_nonce(auth_id.clone(), |_| call.clone())
+						{
+							Ok(_) => guard.forget(),
+							Err(e) => {
+								log::error!("Failed to send a dispatchable transaction: {:?}", e)
+							},
+						}
+					},
+					Err(FinishedTask) => {
+						log::debug!("Already handled Task ({:?}, {:?}) {task:?}", deadline, id);
+						guard.forget();
+						continue;
+					},
+					Err(Evaluation(cause)) => {
+						log::warn!("Failed to verify pending task {:?} : {:?}", task, cause);
```

Comment:
Is this just a placeholder for now? (If not, where will we submit the extrinsic to fail the task?)

Reply:
The TaskV2 implementor is responsible for generating a Call to persist the task's output. If you wish to persist a failure, and for that matter, complete the task with a persisting failure, you are free to do so while still being able to throw an Evaluation error to retry the task later, i.e. the evaluation error is not fatal.